hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndexUtils.java

```diff
@@ -177,27 +172,27 @@ public static <R> HoodieRecord<R> tagRecord(HoodieRecord<R> record, HoodieRecord
    * @param candidateRecordKeys - Candidate keys to filter
    * @return List of pairs of candidate keys and positions that are available in the file
    */
-  public static List<Pair<String, Long>> filterKeysFromFile(Path filePath, List<String> candidateRecordKeys,
+  public static List<Pair<String, Long>> filterKeysFromFile(Path filePath, Set<String> candidateRecordKeys,
                                                             Configuration configuration) throws HoodieIndexException {
+    if (candidateRecordKeys.isEmpty()) {
```

Comment:
Early exit if there are no candidate records. Previously we were creating a file reader before exiting.

Reply:
+1